### PR TITLE
Add URL and username to ES connection error log

### PR DESCRIPTION
**Pull request by ShahabT**
**What changed?**
Added URL and username to an error log

**Why?**
For better debugging experience

**How did you test it?**
No logic change

**Potential risks**
None

**Is hotfix candidate?**
No

### CODE REVIEW

File: temporal/fx.go [hunk 1]
```diff
@@ -258,7 +258,8 @@ func ServerOptionsProvider(opts []ServerOption) (serverOptionsProvider, error) {
 
 		esClient, err = esclient.NewClient(esConfig, esHttpClient, logger)
 		if err != nil {
-			return serverOptionsProvider{}, fmt.Errorf("unable to create Elasticsearch client: %w", err)
+			return serverOptionsProvider{}, fmt.Errorf("unable to create Elasticsearch client (URL = %v, username = %q): %w",
+				esConfig.URL, esConfig.Username, err)
```
Comment (bergundy) [inline temporal/fx.go:262]: Is there any risk of leaking sensitive credentials this way?

**Review by rodrigozhou — APPROVED** [review]

File: temporal/fx.go (hunk 1 above)
Comment (rodrigozhou) [reply temporal/fx.go:262]: Hm. I think URL and username should be fine.
cc: @yiminc

Comment (ShahabT) [reply temporal/fx.go:262]: @bergundy URL does not hold a password (password is in esConfig) the URL.String() output will be in the following format, which seems safe to me:
`scheme://userinfo@host/path?query#fragment`

Comment (ShahabT) [reply temporal/fx.go:262]: oh actually, maybe not lm check again

Comment (ShahabT) [reply temporal/fx.go]: Updated it to `Redacted`

**Pull request merged**